File: CPP/ThreadTest.cpp
#include <iostream>
#include <thread>
//#include <unistd.h>
#include <string>
#include <chrono>
using namespace std;

void outThread(string str){
	for (int x=0; x < 30; x++){
		cout << str << endl;
		this_thread::sleep_for(chrono::seconds(1));
		//sleep(1);
	}
}

int main(){
	
	thread out1 (outThread, "HelloThere");

	while (true){
		string inp;
		cout << "what: ";
		getline( cin, inp);
		//cin >> inp;
		if (inp == "x"){
			break;
		} else{
			cout << "yep yep yep " << inp << endl;
		}
	}

	out1.join();

	return 0;
}

File: Python/ThreadTest.py
import threading
import time

class output (threading.Thread):
	def __init__(self, out):
		threading.Thread.__init__(self)
		self.outputString = out

	def run(self):
		for x in range(30):
			print(self.outputString)
			time.sleep(1)



#thread.start_new_thread( output, "hello" )

out1 = output("hello")
out1.start()

while 1:
	inp = input("What?")
	if inp == 'x':
		break
	else:
		print("yep yep yep" + inp)
		#print(inp)
